caikit_nlp/data_model/text.py

```diff
@@ -0,0 +1,27 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""These interfaces can be promoted to caikit/caikit for wider usage
+when applicable to multiple modules
+"""
+
+# First Party
+from caikit.core import DataObjectBase
+import caikit
+
+
+@caikit.core.dataobject(package="caikit_data_model.caikit_nlp")
+class Span(DataObjectBase):
+    start: int
+    end: int
```

Comment:
nit: can we add a comment if the `end` is inclusive  or would you say thats implementation detail ?

Reply:
I think that's a bit of an implementation detail since even if we write a comment here, this does not enforce whether or not it's inclusive. Best practice may be for it to not be inclusive, but users of this data model are certainly free to do whatever they want.